src/components/CheckAuth/CheckAuth.js

```diff
@@ -21,17 +21,35 @@ import * as React from 'react';
 import { AppState } from 'react-native';
 import styled from 'styled-components/native';
 import { connect } from 'react-redux';
-import get from 'lodash.get';
 
 import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
 import { DECRYPTING, INVALID_PASSWORD, GENERATING_CONNECTIONS } from 'constants/walletConstants';
 import { checkAuthAction } from 'actions/authActions';
-import { Container } from 'components/Layout';
+import { Container, Wrapper } from 'components/Layout';
 import Loader from 'components/Loader';
 import ErrorMessage from 'components/ErrorMessage';
 import PinCode from 'components/PinCode';
 import { addAppStateChangeListener, removeAppStateChangeListener } from 'utils/common';
-import { getKeychainDataObject } from 'utils/keychain';
+import { getPrivateKey } from 'utils/keychain';
+import SlideModal from 'components/Modals/SlideModal';
+import Header from 'components/Header';
+
+type HeaderProps = {
+  title?: String,
+  centerTitle?: boolean,
+  onClose?: Function,
+  onBack?: Function,
+}
+
+type ModalProps = {
+  onModalHide: Function,
+  onModalHidden?: Function,
+  title?: string,
+  centerTitle?: boolean,
+  fullScreen?: boolean,
+  showHeader?: boolean,
+  isVisible?: boolean
+}
```

Comment:
I feel like we can just hide most of these props inside the component's props, now that's gonna have it's own modal. It would help reduce noise, and make the modal always look the same. It doesn't seem like we need to expose the modal to the outside, so we could have `onModalHide` -> `onHide`, and so.

What do you think? :thinking: